### PR TITLE
Fix ETCD_ACTIVE_ENDPOINTS environment variable not found issue

### PR DESCRIPTION
**Pull request by arloliu**
**Description of the change**
Fix the issue created by PR https://github.com/bitnami/bitnami-docker-etcd/pull/42
Create a `setup_etcd_active_endpoints` function to  set `ETCD_ACTIVE_ENDPOINTS` env.

And call this function in `get_member_id` function.

**Benefits**

Fix etcd startup issue if `ETCD_DISABLE_STORE_MEMBER_ID` set to `yes`

**Possible drawbacks**
None

**Applicable issues**

**Additional information**
PR https://github.com/bitnami/bitnami-docker-etcd/pull/42

### CODE REVIEW

**Comment by haorenfsa** [comment]
Hi, it seems that you're upgrading from a previous release with `ETCD_DISABLE_STORE_MEMBER_ID` set to `yes`.
Unfortunately, current image does not support upgrading from previous release. I'm still working on it see: issue #46.

**Comment by haorenfsa** [comment]
This patch alone looks good to me, though.

**Comment by arloliu** [comment]
thanks, I think disabling the member id store and letting the pod to re-join itself is a better approach than managing the member id in each pod.

File: 3.3/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh [hunk 1]
```diff
@@ -425,6 +424,21 @@ is_healthy_etcd_cluster() {
...
+########################
+# Checks if there are enough active members, will also set ETCD_ACTIVE_ENDPOINTS
+# Globals:
+#   ETCD_*
+# Arguments:
+#   None
+# Returns:
+#   Boolean
+########################
+is_healthy_etcd_cluster() {
```
Comment (nschimke) [inline 3.3/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh:439]: This function references $cluster_size which is undefined here after this change.

Comment (arloliu) [reply 3.3/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh:439]: Thanks a lot, I fixed it, please help to review

**Comment by arloliu** [comment]
Sure,  I created a local docker image `etcd:test2`, set it to etcd helm chart with the following options:
1. set env var `ETCD_DISABLE_STORE_MEMBER_ID` to `yes`
2. set `removeMemberOnContainerTermination` to `false` (It will not call `preStop` script that removes member from cluster)

And I tested 2 cases
1. Delete `/bitnami/etcd/data/member_id` manually before rolling restart etcd statefulset
2. Keep `/bitnami/etcd/data/member_id`

The line of log below indicating the `get_member_id` function fetch member if from `etcdctl member list` successfully and it doesn't have unbound `ETCD_ACTIVE_ENDPOINTS` issue
```
etcd 20:26:28.37 INFO  ==> member id: 45a18acb10aa275e
```
**Stateful screenshot**
image

**Pod log**
image

File: 3.3/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh [hunk 2]
```diff
@@ -748,6 +763,12 @@ get_member_id() {
     fi
     local ret
     local -a extra_flags
+
+    local etcd_active_endpoints=${ETCD_ACTIVE_ENDPOINTS:-}
+    if [ -z ${etcd_active_endpoints} ]; then
```
[the remark this answers is not in the document]
Comment (arloliu) [reply 3.3/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh:768]: ok, let me test if is_empty_value works

**Comment by haorenfsa** [comment]
Looks good! @arloliu So have you succeeded in upgrading from earlier release to disabled member_id store & prestop with this patch?

File: 3.5/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh [hunk 3]
```diff
@@ -689,7 +689,7 @@ etcd_initialize() {
                 replace_in_file "$ETCD_NEW_MEMBERS_ENV_FILE" "^" "export "
                 etcd_store_member_id
             elif ! is_empty_value "$member_id"; then
-                info "Updating member in existing cluster"
```
Comment (javsalgar) [inline 3.5/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh:677]: Same here

**Review by javsalgar — CHANGES_REQUESTED** [review]
Small change requested!

File: 3.4/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh [hunk 4]
```diff
@@ -689,7 +689,7 @@ etcd_initialize() {
                 replace_in_file "$ETCD_NEW_MEMBERS_ENV_FILE" "^" "export "
                 etcd_store_member_id
             elif ! is_empty_value "$member_id"; then
-                info "Updating member in existing cluster"
```
[the remark this answers is not in the document]
Comment (arloliu) [reply 3.4/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh]: updated

File: 3.5/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh (hunk 3 above)
Comment (arloliu) [reply 3.5/debian-11/rootfs/opt/bitnami/scripts/libetcd.sh]: updated

**Review by javsalgar — APPROVED** [review]

**Pull request merged**